crates/nargo/src/cli/verify_cmd.rs

```diff
@@ -53,18 +53,18 @@ pub fn verify_with_path<P: AsRef<Path>>(
             read_inputs_from_file(curr_dir, VERIFIER_INPUT_FILE, Format::Toml, public_abi)?;
     }
 
```

Comment:
Not related to this PR:

Note that the mapping from AbiMap to WitnessMap assumes that the map is injective (1to1). This is not the case, if you can make a public input become a public output. 

To expand consider:

```rust
    fn main(x : pub Field) -> pub Field {
        x
    }
```

We have a public input which is also a public output. The ACIR would look like this:

```rust
public_input : [1]
public_output : [1]
```

The `1` being in public_output lets us know that the witness with index 1 is a public output.

This resembles the case:

```rust
    fn main(x : Field) -> pub Field {
        x
    }
```

The above can be disallowed, so in nargo, we atleast know that this was not the case.

So the only other case that we need to disambiguate is the following:

```rust
fn main(x : Field) -> pub Field {
   let y = x + x;
   y
}
```

Let assume `y` gets the witness index `2`. Then the ACIR will look like:

```
public_input : [2]
public_output : [2]
```

The only way to disambiguate this case from the first case, is to know whether the variable was apart of the ABI or if it was not.

If the variable was a part of the ABI, then the only way to get:

```
public_input : [2]
public_output : [2]
```

Is if `2` was also public. Since as noted, we do not allow a private abi variable to be returned as public.

If the variable was not a part of the ABI, then the only way to get:

```
public_input : [2]
public_output : [2]
```

Is if the variable was a private variable not a part of the ABI

Reply:
Not following this 100%, will give more thoughts later.